sdk/rust/src/client.rs

```diff
@@ -59,88 +59,57 @@ fn init_params(payload: &Map<String, Value>, appid: &str, apikey: &str) -> Value
+    let client = ClientBuilder::new()
+                .danger_accept_invalid_certs(SKIP_CERTS)
                 .build()?;
 
-            let resp = client
-                .post(&format!("{}CreateKey", self.base_url))
-                .header("Content-Type", "application/json")
-                .body(body.to_string())
-                .send()
-                .await?;
-
-            if !resp.status().is_success() {
-                //return Err(Error::from(resp.status()));
-                println!("send request failed!");
-            }
-
-            let resp_json: Value = serde_json::from_str(&resp.text().await?)?;
-            let keyid = resp_json["result"]["keyid"]
-                    .as_str()
-                    .ok_or_else(|| anyhow!("Missing keyid"))?
-                    .to_owned();
-        
-            Ok(keyid)
-        }
-
-    async fn decrypt(&mut self, _keyid: &str, _data: &str, _aad: Option<&str>) -> Result<String>{
-        Ok("Not Implemented".to_string())
-    }
-
-    async fn encrypt(&mut self, _keyid: &str, _data: &str, _aad: Option<&str>) -> Result<String>{
-        Ok("Not Implemented".to_string())
-    }
-}
-
-#[async_trait]
-impl Secret for EHSMClient {
-
-    #[warn(unused_variables)]
-    async fn create_secret(
-        &mut self,
-        secret_name: &str,
-        secret_data: &str,
-        encryption_key_id: Option<&str>,
-        description: Option<&str>,
-        rotation_interval: Option<u32>) -> Result<String> {
-            Ok("Not Implemented".to_string())
+        let resp = client
+            .post(&format!("{}/ehsm?Action={}", base_url, action))
+            .header("Content-Type", "application/json")
+            .body(body.to_string())
+            .send()
+            .await?;
+    
+        if !resp.status().is_success() {
+            println!("send request failed!");
         }
+        let resp_json: Value = serde_json::from_str(&resp.text().await?)?;
+        Ok(resp_json)
 }
 
 impl EHSMClient {
-    pub fn new(base_url: &str, appid: &str, apikey: &str) -> Self {
+    pub fn new() -> Self {
         Self {
-            base_url: base_url.to_string(),
-            appid: appid.to_string(),
-            apikey: apikey.to_string(),
+            base_url: env::var("EHSM_ADDR").expect("Please export EHSM_ADDR."),
+            appid: env::var("EHSM_APPID").expect("Please export EHSM_APPID."),
+            apikey: env::var("EHSM_APIKEY").expect("Please export EHSM_APIKEY."),
         }
     }
+
+    pub async fn enroll(base_url: &str) {
+        let client = ClientBuilder::new()
+                .danger_accept_invalid_certs(SKIP_CERTS)
+                .build()
+                .unwrap();
+    
+        let resp = client
+                .get(&format!("{}Enroll", base_url.to_string()))
+                .header("Content-Type", "application/json")
+                .body("")
+                .send()
+                .await
```

Comment:
why not call do_action directly?

Reply:
do_ action is used in KMS to send "Post" requests, and only enroll sends "Get" requests.

Reply:
there're some other interfaces will use GET method, so do we need to wrap another method such as do_get.
and to keep align seems you need to update do_action to do_post?

Reply:
fix